### PR TITLE
Use Json by default instead of Oj

### PR DESCRIPTION
**Pull request by miry**
Oj is not supported Jruby. That's why better to use MultiJson and if you need to add performance do next in initializers:

```ruby
MultiJson = Oj
```

### CODE REVIEW

**Comment by driv3r** [comment]
:+1:

**Comment by driv3r** [comment]
although we could add `if defined?(Oj)` and then set it up by default, what do you think?

**Comment by miry** [comment]
@driv3r good idea. will add this.

File: lib/hasoffersv3/response.rb [hunk 1]
```diff
@@ -1,11 +1,10 @@
...
-
 class HasOffersV3
   class Response
     attr_reader :body, :http_status_code, :http_message, :http_headers
 
-    def initialize(response)
-      @body             = Oj.load(response.body.to_s)
+    def initialize(response, configuration=::HasOffersV3::Configuration.new)
+      @json_adapter     = configuration.json_adapter
+      @body             = json_adapter.load(response.body.to_s)
```
Comment (driv3r) [inline lib/hasoffersv3/response.rb]: maybe it's easier to just pass the adapter, no?

File: lib/hasoffersv3/response.rb [hunk 2]
```diff
@@ -74,5 +71,14 @@ def get_error_values(obj)
         obj.map { |error| error["err_msg"] || error["publicMessage"] }
       end
     end
+
+    def self.json_adapter
+      @json_adapter ||= (defined?(Oj) ? Oj : (defined?(MultiJson) ? MultiJson : JSON))
```
Comment (driv3r) [inline lib/hasoffersv3/response.rb]: I would split it into regular if/else, as multiple inline ternary operators read bad

**Comment by driv3r** [comment]
:+1:

**Pull request merged**